src/lib.rs

```diff
@@ -285,6 +291,8 @@ impl LightningNode {
 
 impl Drop for LightningNode {
     fn drop(&mut self) {
+        self.stop_chain_sync.store(true, Ordering::Release);
```

Comment:
Since the chain sync runs in an other thread, I think we have to actually wait for the thread to stop. Just sending the killing signal won't ensure a graceful shutdown of the sync before the application is terminated.

Instead of just setting an AtomicBool, we should use something that blocks the current thread until the thread has actually received the shutdown signal. Check out https://stackoverflow.com/questions/26199926/how-to-terminate-or-suspend-a-rust-thread-from-another-thread